examples/react/virtualized-rows/src/fixedHeightTable.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useVirtualizer } from '@tanstack/react-virtual'
+
+import { Person } from './makeData'
+
+type FixedHeightTableProps = {
+  // The data to render
+  data: any
+  // The columns to render
+  columns: ColumnDef<Person>[]
+  // The height of the table
+  // @default 500px
+  height?: number
+}
+
+/**
+ * Renders fixed height virtualised table
+ */
+export function FixedHeightTable({
+  data,
+  columns,
+  height = 500,
+}: FixedHeightTableProps) {
+  const [sorting, setSorting] = React.useState<SortingState>([])
+
+  const table = useReactTable({
+    data,
+    columns,
+    state: {
+      sorting,
+    },
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    debugTable: true,
+  })
+
+  const tableContainerRef = React.useRef<HTMLDivElement>(null)
+
+  const { rows } = table.getRowModel()
+  const rowVirtualizer = useVirtualizer({
+    count: rows.length,
+    getScrollElement: () => tableContainerRef.current,
+    estimateSize: () => 54,
+    overscan: 10,
+  })
+
+  const virtualRows = rowVirtualizer.getVirtualItems()
+  const totalSize = rowVirtualizer.getTotalSize()
+
+  // This is where the magic happens, essentially create a large row with height of total rows
+  // before and after the first/last displayed row so the scroll bar works correctly
+  const paddingTop = virtualRows.length > 0 ? virtualRows?.[0]?.start || 0 : 0
+  const paddingBottom =
+    virtualRows.length > 0
+      ? totalSize - (virtualRows?.[virtualRows.length - 1]?.end || 0)
+      : 0
```

Comment:
We can keep those values together

```suggestion
  const [paddingTop, paddingBottom] =
    virtualRows.length > 0
      ? [
          Math.max(0, virtualRows[0].start),
          Math.max(0, rowVirtualizer.getTotalSize() - virtualRows[virtualRows.length - 1].end),
        ]
      : [0, 0]
```

Reply:
Updated but had to add some extra logic to remove undefined errors in TS